src/clj_foundation/grep.clj

```diff
@@ -0,0 +1,146 @@
+(ns clj-foundation.grep
+  "Unix-style grep for arbitrary Clojure nested collections."
+  (:require [clojure.zip :as zip]
+            [clojure.set :as set]
+            [clj-foundation.tree-visit :as v]
+            [clj-foundation.patterns :refer [nothing]])
+  (:import [java.util Map Map$Entry]))
+
+
+(defn- result-coll?
+  "Don't return Map$Entrys as results; other collection types are supported."
+  [node]
+  (and (not (instance? Map$Entry node))
+       (or (map? node) (vector? node) (list? node) (set? node) (seq? node))))
+
+
+(defn- traversable-coll?
+  "Is this a kind of collection we can traverse?"
+  [node]
+  (or (map? node) (vector? node) (list? node) (set? node) (seq? node)))
+
+
+(defn root-node?
```

Comment:
I might consider adding metadata for functions with return type of $zipObject throughout the library.